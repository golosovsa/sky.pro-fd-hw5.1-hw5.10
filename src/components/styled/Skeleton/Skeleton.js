import styled, { keyframes } from "styled-components";
import { layoutSizeMixin, calcWidth } from "../../../globals/layout";
import { applyThemeMixin } from "../../../globals/theme";

const shineDuration = "5s";

const Skeleton = styled.div`
    ${applyThemeMixin("main", "sceleton")}
    ${(props) => layoutSizeMixin(props.width || "100", props.height || "100")}
    animation: ${(props) => keyframes`
        0%{
            background-position: ${calcWidth(
                -1920 - (Number(props.animationOffset) || 0)
            )};
        }
        100% {
            background-position: ${calcWidth(
                1920 * 2 - (Number(props.animationOffset) || 0)
            )};
        }
    `} ${shineDuration} infinite linear;
    background-size: ${calcWidth(1920 * 3)} 1px;
`;

export default Skeleton;
