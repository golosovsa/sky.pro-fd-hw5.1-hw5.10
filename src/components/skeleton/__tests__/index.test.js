import ShallowRenderer from 'react-test-renderer/shallow'
import Skeleton from '..'

describe('<Skeleton />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<Skeleton />)

      expect(tree).toMatchSnapshot()
    })

    it('should render successfully and match snapshot is full', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<Skeleton full={true} animationOffset={100} />)

      expect(tree).toMatchSnapshot()
    })

    it('should render successfully and match snapshot is width and/or height', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(<Skeleton width={200} height={200} />)

      expect(tree).toMatchSnapshot()
    })
  })
})
