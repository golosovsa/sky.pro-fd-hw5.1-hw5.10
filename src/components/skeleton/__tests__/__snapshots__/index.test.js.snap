// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> Snapshot test should render successfully and match snapshot 1`] = `
<styled.div
  animationOffset={0}
  fill="true"
  shineDuration="3s"
/>
`;

exports[`<Skeleton /> Snapshot test should render successfully and match snapshot is full 1`] = `
<styled.div
  animationOffset={100}
  full="true"
  shineDuration="3s"
/>
`;

exports[`<Skeleton /> Snapshot test should render successfully and match snapshot is width and/or height 1`] = `
<styled.div
  animationOffset={0}
  fill="true"
  height={200}
  shineDuration="3s"
  width={200}
/>
`;
